factorial() won't accept floats in python >=3.10

i added python 3.10 to my test matrices and i get failures due to `poppy`. the specific error is:
```
        if _is_odd(n - m):
            return 0
        else:
            for k in range(int((n - m) / 2) + 1):
                coef = ((-1) ** k * factorial(n - k) /
>                       (factorial(k) * factorial((n + m) / 2. - k) * factorial((n - m) / 2. - k)))
E               TypeError: 'float' object cannot be interpreted as an integer

../py310-alldeps/lib/python3.10/site-packages/poppy/zernike.py:174: TypeError
```
`factorial()` won't accept floats as of python 3.10: https://docs.python.org/3/library/math.html#math.factorial.

--- a/.github/workflows/ci_workflows.yml
+++ b/.github/workflows/ci_workflows.yml
@@ -16,7 +16,7 @@ jobs:
 
           - name: Coverage test in Python 3
             os: ubuntu-latest
-            python: 3.10
+            python: '3.10'
             toxenv: py310-syn-cov
 
           - name: Check for Sphinx doc build errors
@@ -36,7 +36,7 @@ jobs:
 
           - name: Try latest versions of all dependencies
             os: ubuntu-latest
-            python: 3.10
+            python: '3.10'
             toxenv: py310-latest-test
 
           - name: Try minimum supported versions

--- a/.github/workflows/ci_workflows.yml
+++ b/.github/workflows/ci_workflows.yml
@@ -16,33 +16,33 @@ jobs:
 
           - name: Coverage test in Python 3
             os: ubuntu-latest
-            python: 3.9
-            toxenv: py39-syn-cov
+            python: 3.10
+            toxenv: py310-syn-cov
 
           - name: Check for Sphinx doc build errors
             os: ubuntu-latest
-            python: 3.8
+            python: 3.9
             toxenv: docbuild
 
           - name: Check accelerated math version
             os: ubuntu-latest
-            python: 3.8
-            toxenv: py38-numexpr-mkl-cov
+            python: 3.9
+            toxenv: py39-numexpr-mkl-cov
 
           - name: Try Astropy development version
             os: ubuntu-latest
-            python: 3.8
-            toxenv: py38-astropydev-test
+            python: 3.9
+            toxenv: py39-astropydev-test
 
           - name: Try latest versions of all dependencies
             os: ubuntu-latest
-            python: 3.9
-            toxenv: py39-latest-test
+            python: 3.10
+            toxenv: py310-latest-test
 
           - name: Try minimum supported versions
             os: ubuntu-latest
-            python: 3.7
-            toxenv: py37-legacy-test
+            python: 3.8
+            toxenv: py38-legacy-test
 
     steps:
     - name: Checkout code

--- a/.github/workflows/publish-to-pypi.yml
+++ b/.github/workflows/publish-to-pypi.yml
@@ -5,22 +5,12 @@ on:
     types: [released]
 
 jobs:
-  build:
-      name: Publish release to PyPI
-      env:
-        PYPI_USERNAME_STSCI_MAINTAINER: ${{ secrets.PYPI_USERNAME_STSCI_MAINTAINER }}
-        PYPI_PASSWORD_STSCI_MAINTAINER: ${{ secrets.PYPI_PASSWORD_STSCI_MAINTAINER }}
-        PYPI_USERNAME_OVERRIDE: ${{ secrets.PYPI_USERNAME_OVERRIDE }}
-        PYPI_PASSWORD_OVERRIDE: ${{ secrets.PYPI_PASSWORD_OVERRIDE }}
-        PYPI_TEST: ${{ secrets.PYPI_TEST }}
-        INDEX_URL_OVERRIDE: ${{ secrets.INDEX_URL_OVERRIDE }}
-      runs-on: ubuntu-latest
-      steps:
-
-          # Check out the commit containing this workflow file.
-          - name: checkout repo
-            uses: actions/checkout@v2
-
-          - name: custom action
-            uses: spacetelescope/action-publish_to_pypi@master
-            id: custom_action_0
+  publish:
+    uses: spacetelescope/action-publish_to_pypi/.github/workflows/workflow.yml@master
+    with:
+      test: false
+      build_platform_wheels: false # Set to true if your package contains a C extension
+    secrets:
+      user: ${{ secrets.PYPI_USERNAME_STSCI_MAINTAINER }}
+      password: ${{ secrets.PYPI_PASSWORD_STSCI_MAINTAINER }} # WARNING: Do not hardcode secret values here! If you want to use a different user or password, you can override this secret by creating one with the same name in your Github repository settings.
+      test_password: ${{ secrets.PYPI_PASSWORD_STSCI_MAINTAINER_TEST }}

--- a/docs/dev_notes.rst
+++ b/docs/dev_notes.rst
@@ -28,6 +28,10 @@ Release Process
 Releasing a new version through AstroConda
 ------------------------------------------
 
+.. admonition:: **Consider this section deprecated as of version 1.0.3!**
+
+ AstroConda is currently limited to Python <=3.7, while POPPY only supports Python >=3.8 as of v1.0.3.
+
 Do this after you've done the above.
 
 To test that an Astroconda package builds, you will need ``conda-build``::

--- a/docs/installation.rst
+++ b/docs/installation.rst
@@ -1,15 +1,13 @@
 Installation
 ==================
 
-POPPY may be installed one of three different ways.
+POPPY may be installed one of two different ways.
 
-1. Using ``conda`` through the `AstroConda channel <https://astroconda.readthedocs.io/en/latest/>`__. This is the recommended channel for most users on MacOS and Linux. But note that AstroConda does not support Windows.
-
-2. Using PyPi in the usual manner for Python packages::
+1. Using PyPI in the usual manner for Python packages::
 
     % pip install poppy --upgrade
 
-3. Cloning the source code hosted in `this repository on GitHub <https://github.com/spacetelescope/poppy>`_. It is possible to directly install the latest development version using your locally installed ``git`` package::
+2. Cloning the source code hosted in `this repository on GitHub <https://github.com/spacetelescope/poppy>`_. It is possible to directly install the latest development version using your locally installed ``git`` package::
 
     % git clone https://github.com/spacetelescope/poppy.git
     % cd poppy

--- a/docs/installation.rst
+++ b/docs/installation.rst
@@ -19,7 +19,7 @@ POPPY may be installed one of three different ways.
 Requirements
 --------------
 
-* Python 3.7, or more recent.
+* Python 3.8, or more recent.
 * The standard Python scientific stack: :py:mod:`numpy`, :py:mod:`scipy`,
   :py:mod:`matplotlib`
 * POPPY relies upon the `astropy

--- a/docs/relnotes.rst
+++ b/docs/relnotes.rst
@@ -5,6 +5,24 @@ Release Notes
 
 For a list of contributors, see :ref:`about`.
 
+1.0.3
+-----
+
+.. _rel1.0.3:
+
+*2022 June 30*
+
+This is another incremental release that introduces support and makes internal adjustments for Python 3.10.
+
+**Other enhancements and fixes:**
+ * zernike.py factorial() calls were corrected to ensure internal values are integers in Python 3.10 (:pr:`493` by :user:`AarynnCarter`)
+
+**Software Infrastructure Updates and Internals:**
+ * Fix for Astropy Development CI test (:pr:`498` by :user:`shanosborne`)
+ * Increase all CI test versions by 1, removing Python 3.7 and adding Python 3.10 (:pr:`503` by :user:`shanosborne`)
+ * Accept various dependabot updates to recommended (not minimum) dependency versions (various PRs by :user:`shanosborne` and :user:`mperrin`)
+ * Updated installation and release instructions to move away from AstroConda (:pr:`509` by :user:`ojustino`)
+
 1.0.2
 -----
 
@@ -37,7 +55,7 @@ This is a very minor re-release, to fix some documentation formatting and releas
 
 *2021 December 7*
 
-This is a major release with significant enhancements and changes, in particular with regards to changes in wavefront sign convention representations. 
+This is a major release with significant enhancements and changes, in particular with regards to changes in wavefront sign convention representations.
 
 .. admonition:: Changes and Clarifications in Signs for Wavefront Error and Phase
 
@@ -51,8 +69,8 @@ This is a major release with significant enhancements and changes, in particular
     changes in the signs of complex exponentials in some Fourier propagation calculations. Depending on your use case this may result in some changes in output PSFs or
     different signs or orientations from prior results.
 
-    See `Sign Conventions for Coordinates, Phase, and Wavefront Error <https://poppy-optics.readthedocs.io/en/latest/sign_conventions_for_coordinates_and_phase.html>`_ for details, discussion, and demonstration. 
-    
+    See `Sign Conventions for Coordinates, Phase, and Wavefront Error <https://poppy-optics.readthedocs.io/en/latest/sign_conventions_for_coordinates_and_phase.html>`_ for details, discussion, and demonstration.
+
     Many thanks to Derek
     Sabatke (Ball Aerospace); Matthew Bergkoetter, Alden Jurling, and Tom Zielinski (NASA GSFC); and
     Randal Telfer (STScI) for invaluable discussions and aid in getting these
@@ -95,7 +113,7 @@ This is a major release with significant enhancements and changes, in particular
 
 *2021 Feb 11*
 
-This release includes several updated optical element classes, bug fixes, and improved documentation. This is intended as a maintenance release shortly before v 1.0 which will introduce some backwards-incompatible changes. 
+This release includes several updated optical element classes, bug fixes, and improved documentation. This is intended as a maintenance release shortly before v 1.0 which will introduce some backwards-incompatible changes.
 
 **New Functionality:**
  * New OpticalElement classes for ScalarOpticalPathDifference, LetterFAperture, and LetterFOpticalPathDifference. (:pr:`386` by :user:`mperrin`)
@@ -172,7 +190,7 @@ This is a minor release primarily for updates in packaging infrastructure, plus 
    transformation only applies the cosine scaling to the optic's appearance, and does *not* introduce wavefront for
    tilt. (:pr:`329` by :user:`mperrin`)
 
- * **Many improvements to the Continuous Deformable Mirror class**: 
+ * **Many improvements to the Continuous Deformable Mirror class**:
 
     * Enhance model of DM actuator influence functions for more precise subpixel spacing of DM actuators, rather than
       pokes separated by integer pixel spacing. This applies to the 'convolution by influence function' method for
@@ -196,7 +214,7 @@ This is a minor release primarily for updates in packaging infrastructure, plus 
 
 **Software Infrastructure Updates and Internals:**
  * PR :pr:`290` for CompoundOpticalSystem involved refactoring the Wavefront and FresnelWavefront classes to both be child classes of a new abstract base class BaseWavefront. This change should be transparent for most/all users and requires no changes in calling code.
- * PR :pr:`306` for wavelength-independent phase subsequently required refactoring of the optical element display code to correctly handle all cases. As a result the display code internals were clarified and made more consistent. (:pr:`314` and :pr:`321`  by :user:`mperrin` with contributions from :user:`ivalaginja` and :user:`shanosborne`). Again this change should be transparent for users. 
+ * PR :pr:`306` for wavelength-independent phase subsequently required refactoring of the optical element display code to correctly handle all cases. As a result the display code internals were clarified and made more consistent. (:pr:`314` and :pr:`321`  by :user:`mperrin` with contributions from :user:`ivalaginja` and :user:`shanosborne`). Again this change should be transparent for users.
  * Removed deprecated / unused decorator function in WFE classes, making their `get_opd` function API consistent with the rest of poppy. (:pr:`322` by :user:`mperrin`)
  * Accomodate some upstream changes in astropy (:pr:`294` by :user:`shanosborne`, :pr:`330` by :user:`mperrin`)
  * The `poppy.Instrument._get_optical_system` function, which has heretofore been an internal method (private, starting with
@@ -225,7 +243,7 @@ This is a minor release primarily for updates in packaging infrastructure, plus 
 apertures such as from inclined optics. You may see warnings about this in cases when you use `inclination_y` or
 `inclination_x` attributes on a circular aperture. This warning is generally benign; the calculation is still more
 accurate than it would be without the subpixel sampling, though not perfectly precise. This known issue will likely be
-improved upon in a future release. 
+improved upon in a future release.
 
 
 0.8.0

--- a/poppy/__init__.py
+++ b/poppy/__init__.py
@@ -23,7 +23,7 @@ try:
 except ImportError:
     __version__ = ''
 
-__minimum_python_version__ = "3.7"
+__minimum_python_version__ = "3.8"
 
 
 class UnsupportedPythonError(Exception):

--- a/poppy/wfe.py
+++ b/poppy/wfe.py
@@ -144,7 +144,7 @@ class ParameterizedWFE(WavefrontError):
     @utils.quantity_input(coefficients=u.meter, radius=u.meter)
     def __init__(self, name="Parameterized Distortion", coefficients=None, radius=1*u.meter,
                  basis_factory=None, **kwargs):
-        if not isinstance(basis_factory, collections.Callable):
+        if not isinstance(basis_factory, collections.abc.Callable):
             raise ValueError("'basis_factory' must be a callable that can "
                              "calculate basis functions")
         self.radius = radius

--- a/poppy/zernike.py
+++ b/poppy/zernike.py
@@ -84,7 +84,7 @@ def str_zernike(n, m):
     terms = []
     for k in range(int((n - m) / 2) + 1):
         coef = ((-1) ** k * factorial(n - k) /
-                (factorial(k) * factorial((n + m) / 2. - k) * factorial((n - m) / 2. - k)))
+                (factorial(k) * factorial(int((n + m) / 2) - k) * factorial(int((n - m) / 2) - k)))
         if coef != 0:
             formatcode = "{0:d}" if k == 0 else "{0:+d}"
             terms.append((formatcode + " r^{1:d} ").format(int(coef), n - 2 * k))
@@ -171,7 +171,7 @@ def R(n, m, rho):
     else:
         for k in range(int((n - m) / 2) + 1):
             coef = ((-1) ** k * factorial(n - k) /
-                    (factorial(k) * factorial((n + m) / 2. - k) * factorial((n - m) / 2. - k)))
+                    (factorial(k) * factorial(int((n + m) / 2) - k) * factorial(int((n - m) / 2) - k)))
             output += coef * rho ** (n - 2 * k)
         return output
 

--- a/poppy/zernike.py
+++ b/poppy/zernike.py
@@ -420,7 +420,7 @@ def zernike_basis_faster(nterms=15, npix=512, outside=np.nan):
         else:
             for k in range(int((n - m) / 2) + 1):
                 coef = ((-1) ** k * factorial(n - k) /
-                        (factorial(k) * factorial((n + m) / 2. - k) * factorial((n - m) / 2. - k)))
+                        (factorial(k) * factorial(int((n + m) / 2) - k) * factorial(int((n - m) / 2) - k)))
                 output += coef * rho ** (n - 2 * k)
             return output
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 astropy==5.0.4
 matplotlib==3.5.1
 numpy==1.22.4
-scipy==1.8.0
+scipy==1.8.1

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
-astropy==5.0.2
+astropy==5.0.4
 matplotlib==3.5.1
 numpy==1.22.2
 scipy==1.8.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 astropy==5.0
 matplotlib==3.5.1
-numpy==1.22.1
+numpy==1.22.2
 scipy==1.7.3

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 astropy==5.0.4
 matplotlib==3.5.1
-numpy==1.22.3
+numpy==1.22.4
 scipy==1.8.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
-astropy==5.0
+astropy==5.0.2
 matplotlib==3.5.1
 numpy==1.22.2
 scipy==1.7.3

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
-astropy==5.0.4
+astropy==5.1
 matplotlib==3.5.2
 numpy==1.22.4
 scipy==1.8.1

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 astropy==5.0.2
 matplotlib==3.5.1
 numpy==1.22.2
-scipy==1.7.3
+scipy==1.8.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 astropy==5.0.4
 matplotlib==3.5.1
-numpy==1.22.2
+numpy==1.22.3
 scipy==1.8.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 astropy==5.0.4
-matplotlib==3.5.1
+matplotlib==3.5.2
 numpy==1.22.4
 scipy==1.8.1

--- a/setup.cfg
+++ b/setup.cfg
@@ -23,7 +23,7 @@ install_requires =
     scipy>=1.5.0
     matplotlib>=3.2.0
     astropy>=4.0.0
-python_requires = >=3.7
+python_requires = >=3.8
 setup_requires = setuptools_scm
 
 [options.extras_require]

--- a/tox.ini
+++ b/tox.ini
@@ -17,7 +17,7 @@ deps=
     legacy: numpy==1.18.*
     legacy: astropy==4.0.*
     latest: -rrequirements.txt
-    astropydev: git+git://github.com/astropy/astropy
+    astropydev: git+https://github.com/astropy/astropy
     numexpr: numexpr>=2.0.0
 conda_deps=
     mkl: mkl_fft

--- a/tox.ini
+++ b/tox.ini
@@ -1,9 +1,9 @@
 [tox]
 envlist =
-    py{37,38,39}-test
-    py{37,38,39}-{astropydev,legacy,latest}-test
-    py{37,38,39}{,-syn}-cov
-    py38-numexpr-mkl-cov
+    py{38,39,310}-test
+    py{38,39,310}-{astropydev,legacy,latest}-test
+    py{38,39,310}{,-syn}-cov
+    py39-numexpr-mkl-cov
 
 [testenv]
 passenv = *
@@ -31,7 +31,7 @@ commands=
     cov: codecov -F -e TOXENV
 
 [testenv:docbuild]
-basepython= python3.8
+basepython= python3.9
 deps=
     sphinx
     sphinx_rtd_theme
@@ -48,7 +48,7 @@ commands=
     sphinx-build docs docs/_build
 
 [testenv:codestyle]
-basepython= python3.8
+basepython= python3.9
 skip_install = true
 description = check package code style
 deps =

